lib/msal-node/src/config/Configuration.ts

```diff
@@ -228,20 +240,42 @@ export function buildManagedIdentityConfiguration({
         managedIdentityIdParams
     );
 
-    const systemOptions: Required<NodeSystemOptions> = {
-        ...DEFAULT_SYSTEM_OPTIONS,
-        loggerOptions: system?.loggerOptions || DEFAULT_LOGGER_OPTIONS,
-        networkClient: new HttpClient(
+    const loggerOptions: LoggerOptions =
+        system?.loggerOptions || DEFAULT_LOGGER_OPTIONS;
+
+    let networkClient: INetworkModule;
+    // use developer provided network client if passed in
+    if (system?.networkClient) {
+        networkClient = system.networkClient;
+        // otherwise, create a new one
+    } else {
+        networkClient = new HttpClient(
             system?.proxyUrl,
             system?.customAgentOptions as
                 | http.AgentOptions
                 | https.AgentOptions,
             true // Managed Identity
-        ),
-    };
+        );
+    }
+
+    // wrap the network client with a retry policy if the developer has not disabled the option to do so
+    if (!system?.disableInternalRetries) {
```

Comment:
Is this code common to all? Managed Identity and other calls to ests? If yes, is there somewhere this flag is defaulted to true? Are there tests in node that confirm that only 1 call is made? If not can you add a test to confirm there is no retry in the general flow?

Reply:
This is only for the ManagedIdentity configuration. flag is defaulted to false.